addons/payment_mollie/models/payment_transaction.py

```diff
@@ -0,0 +1,126 @@
+        if self.provider != 'mollie':
+            return res
+
+        api_url = False
+        mollie_payload = self._mollie_prepare_payment_payload()
+        payment_data = self.acquirer_id._mollie_make_request('/payments', data=mollie_payload)
+
+        api_url = payment_data["_links"]["checkout"]["href"]
+
+        # We are setting acquirer reference as we are receiving it before the redirection to mollie.
+        # This will help us verify the validity of the transaction later.
+        self.acquirer_reference = payment_data.get('id')
+
+        return {
+            'api_url': api_url
+        }
+
+    def _get_tx_from_feedback_data(self, provider, data):
+        """ Override of payment to find the transaction based on Mollie data.
+
+        :param str provider: The provider of the acquirer that handled the transaction
+        :param dict data: The feedback data sent by the provider
+        :return: The transaction if found
+        :rtype: recordset of `payment.transaction`
+        :raise: ValidationError if the data match no transaction
+        """
+        tx = super()._get_tx_from_feedback_data(provider, data)
+        if provider != 'mollie':
+            return tx
+
+        tx = self.search([('reference', '=', data.get('ref')), ('provider', '=', 'mollie')])
+        if not tx:
+            raise ValidationError(
+                "Mollie: " + _("No transaction found matching reference %s.", data.get('ref'))
+            )
+        return tx
+
+    def _process_feedback_data(self, data):
+        """ Override of payment to process the transaction based on Mollie data.
+
+        Note: self.ensure_one()
+
+        :param dict data: The feedback data sent by the provider
+        :return: None
+        """
+        super()._process_feedback_data(data)
+        if self.provider != 'mollie':
+            return
+
+        payment_data = self.acquirer_id._mollie_make_request(f'/payments/{self.acquirer_reference}', method="GET")
+        payment_status = payment_data.get('status')
+
+        if payment_status == 'pending':
+            self._set_pending()
+        elif payment_status == 'authorized':
+            self._set_authorized()
+        elif payment_status == 'paid':
+            self._set_done()
+        elif payment_status in ['expired', 'canceled', 'failed']:
+            self._set_canceled("Mollie: " + _("Canceled payment with status: %s", payment_status))
+        else:
+            _logger.info("Received data with invalid payment status: %s", payment_status)
+            self._set_error(
+                "Mollie: " + _("Received data with invalid payment status: %s", payment_status)
+            )
+
+    def _mollie_prepare_payment_payload(self):
+        """ This method prepare the payload based in api type.
+
+        Note: this method are splitted so we can write test cases
+
+        :return: data of created record received from mollie api
+        :rtype: dict
+        """
```

Comment:
```suggestion
    def _mollie_prepare_payment_request_payload(self):
        """ Create the payload for the payment request based on the transaction values.

        :return: The request payload
        :rtype: dict
        """
```
No need to specify that it is split for the tests. There is nothing wrong with extracting this method.